### ISSUE TITLE
App is very nonreactive

### ISSUE THREAD

**Issue by MauroMombelli** [Nitrokey/nitrokey-app#227]
When i connect they key, the app takes a bit to show i can open the wallet, and even when ready clicking the button does nothing for many seconds, make very hard to understand what is going on.

Also a couple of time cling the app did nothing, i had to physically reconnect the card. Over gpg and gpg2 te card is very fast

**Comment by szszszsz** [comment]
Hi! 
What OS and app version do you use? You can check the latter in About window.

**Comment by MauroMombelli** [comment]
Arch Linux(it uses last got version iirc), with a nitrokey pro fw 0.8

**Comment by szszszsz** [comment]
OK, it looks like it uses 0.6.3 [link](https://aur.archlinux.org/packages/nitrokey-app/)

**Comment by MauroMombelli** [comment]
I confirm, 0.6.3 and fw 0.8.

The info menu need 2-3 seconds to appear (ssd, i7, 16gb ram) and after unlocking the wallet the app didn't react to right click for ~5 seconds.

Gpg --card-status is almost immmediate

**Comment by szszszsz** [comment]
I see latest package is Qt 5.8, so there should not be any issues from this side.
It is probably caused by reading all Password Safe entries (their names to be exact) from the device.

We are currently in the process of releasing Nitrokey App v1.0-beta1. It should be much faster. 
There is no package for Arch Linux yet - would like to test it and compile it from the source?

**Comment by MauroMombelli** [comment]
No problem! Just point out a relatively stable commit and this weekend I'll
test it.

**Comment by szszszsz** [comment]
Great!
Please use this command to download sources:
```
git clone  -b  v1.0-beta.1 --recursive  https://github.com/Nitrokey/nitrokey-app.git
```
It uses C++14, so G++>=5.0 is needed. Building should be as easy as invoking `cmake`. It requires Qt 5.6 and libhidapi.
Please check `README` for instructions. In case of any problems please let me know.

**Comment by MauroMombelli** [comment]
done, much better!

I'm on Plasma 5.9.4, so it should play nice with qt5. at plugin i notice that even before asking for user pin, to load some "password".

Also the right-click menu is different than the left click: the left click close the extension tray below when left-clicking option, but at least will leave with a floating menu where you can actually chose the action; instead if open by righr menu, the left click on option close te app menu but keep the system tray extension open...

i took a couple of screenshot. Basically i expect all to work like right click, but the configure opion to have a > like in the left click that display all the option (or on click to open something istead of just disappear!)

![left click](https://cloud.githubusercontent.com/assets/6372561/24272160/4e8c5a42-101c-11e7-808a-84244c89a779.png)

![right click](https://cloud.githubusercontent.com/assets/6372561/24272172/5d69b8c0-101c-11e7-95f9-bea384c4bc3f.png)

**Comment by MauroMombelli** [comment]
i manage to segfault after connecting/disconnecting the pen multiple time, and changing the admin pin work but give no feedback, like the "ok" button does not work

**Comment by szszszsz** [comment]
Thank you for test! I am glad you like it. I have divided your report to separate issues. 

As for:
> at plugin i notice that even before asking for user pin, to load some "password".

This is not clear to me - could you elaborate? Password menu should be populated by OTP slots on device connection and (after unlocking safe) with Password Safe slots. Is it not working that way?

Regarding the left/right click on KDE tray - both mouse buttons are set to same action - opening tray menu. Is that different action than for other applications? If yes, how do they behave? Would opening `Configuration` window (with left mouse click) be OK?

**Comment by MauroMombelli** [comment]
> Password menu should be populated by OTP slots on device connection and (after unlocking safe) with Password Safe slots. Is it not working that way?

At load i can see a voice in the menu that tell me he is loading the password, very nice so one know he has to wait and can see the % done. Is a bit confusing because it load even if i have nothing there, and even if i don't use the pin; but then i realized OTP are not encripted, so it make sense.

>both mouse buttons are set to same action 

As you can see from the screenshot, the two buttons has two different behavior, and this is i think partially how the tray work; right click keep the extended system tray open, as it is normal it open the menu; but you can see the > in options is missing, and it does not display the submenu!
The left click is ok that close the system tray, as i (and probably also kde team) expect it to open the application window and not some menu. As i guess this system tray api is pretty portable, it should act the same even in other DE. Probably i would open directly the "configuratio" window.

**Comment by szszszsz** [comment]
1) Device has to be queried for OTP slots, each one separately. This is why it takes time and needs progress bar.
2) I see now. A bit strange behaviour to lock sub-menus. What I meant is at application level both mouse buttons execute same action on tray menu. I will fix it or change to show `Configuration` instead (should not this be under double-click though?).

**Comment by MauroMombelli** [comment]
No, tray bar use single click. But now I think about is, there are some
program showing the menu with a left click, like kmix showing the audio
channel. The application might show the list of saved password and left
click on them would copy them, or display an input box if empty. (Just some
random idea I'm having on the spot).

What I realy really miss is an easy setup for: ssh integration, login
integration, login + disk encription integration. I understand this stuff
is really distribution and setup dependent, but maybe something can be
automated.

**Comment by szszszsz** [comment]
I see. We will think about the tray.
As for integrations, we have guides for it: [link](https://www.nitrokey.com/documentation/applications).

As you used the problem here is the need for a cover a multiple distribution configurations and setups.
Automated integration will be a whole new App itself from complication perspective. And even then making an automation is one thing, but maintaining it might be a real pain.

**Comment by szszszsz** [comment]
Closing this issue as it is fixed (and retested) in new App 1.0. 
For other requests please create separate issues.

**Comment by szszszsz** [comment]
Again, thank you for tests and suggestions!
